openedx/course-discovery
# feat: add management command to prepare csv for CSV data loader

### [PROD-2673](https://openedx.atlassian.net/browse/PROD-2673)

### Description

- A temporary management command that provides a fully populated course data csv  provided a partially filled course data csv and product API
- There are some TODOs in the code that will be removed in a followup PR once other data transformers are in.
- **Note**: A follow up PR will be created to include the changes from https://github.com/openedx/course-discovery/pull/3291 once that is merged

### Testing

- Download the partially filled CSV from shared sample data. Be mindful of the CSV headers that should be updated(Verified Price and Start Date)
- Generate Product API's auth token from Postman. Update PRODUCT_API_URL in private.py
- Run the command `./manage.py populate_executive_education_data_csv --auth_token=token --input_csv=input.csv --output_csv=out.csv`
- Verify the correctness of output file
